src/components/Button/Button.js

```diff
@@ -33,17 +31,48 @@ const Button = ({ className, largeText, size, ...other }) => {
   );
 };
 
+/* eslint-disable react/require-default-props */
 Button.defaultProps = {
-  ...defaultProps,
   largeText: null,
+  icon: 'None',
+  kind: 'primary',
+  disabled: false,
+  size: 'default',
 };
 
 Button.propTypes = {
-  ...propTypes,
+  /**
+   * Optional className
+   */
+  className: PropTypes.string,
+
+  /**
+   * Choose an icon type to better communicate what the button does
+   */
+  icon: PropTypes.oneOf(['None', 'Add16', 'Add24', 'Search16', 'Search24']),
+
+  /**
+   * Choose button type
+   */
+  kind: PropTypes.oneOf(['primary', 'secondary', 'danger', 'ghost']),
+
+  /**
+   * Property to disable the button
+   */
+  disabled: PropTypes.bool,
+
+  /**
+   * Icon description for screen readers, if using icons
+   */
+  iconDescription: PropTypes.string,
 
   // It's not possible to add to Carbon's values here, so the `PropType` is recreated to include the large variant.
-  size: oneOf(['default', 'field', 'large', 'small']),
-  largeText: deprecatedProp('size="large"', bool),
+  /**
+   * Choose the size of the button
+   */
+  size: PropTypes.oneOf(['default', 'field', 'large', 'small']),
+
+  largeText: deprecatedProp('size="large"', PropTypes.bool),
```

Comment:
Could you add a description for `largeText` to indicate that it is deprecated, so users can see that message in the storybook prop table? Something short like this would work:

```suggestion
  /**
   * Deprecated prop. Use 'size' instead.
   */
  largeText: deprecatedProp('size="large"', PropTypes.bool),
```